app/watch/page.tsx

```diff
@@ -0,0 +1,483 @@
+"use client"
+import styled from "styled-components"
+import { livestreams } from "../info/livestreams"
+import { PotionBackground } from "../components/PotionBackground"
+import { ErrorBoundary } from "../components/ErrorBoundary"
+
+// Types
+
+interface Talk {
+	speaker: string
+	title: string
+}
+
+interface Livestream {
+	id: string
+	title: string
+	date: string
+	year: number
+	startTime?: string
+	endTime?: string
+	talks: Talk[]
+}
+
+// Functions
+
+const buildYouTubeUrl = (id: string, startTime?: string) => {
+	let url = `https://youtube.com/watch?v=${id}`
+
+	if (startTime && startTime !== "0s") {
+		url += `&t=${startTime}`
+	}
+
+	return url
+}
+
+const generateHeroDescription = (talks: Talk[]) => {
+	return (
+		<>
+			San Diego DevX Monthly Meetup live talks featuring:
+			<br />
+			<br />
+			{talks.map((talk, index) => (
+				<span key={index}>
+					• <SpeakerName>{talk.speaker}</SpeakerName> - <SpeakerName>{talk.title}</SpeakerName>
+					<br />
+				</span>
+			))}
+			<br />
+			Join our discord and see events at https://devx.network/
+		</>
+	)
+}
+
+// Components
+
+export default function Watch() {
+	// Group livestreams by year for sectioned display
+	const videosByYear = livestreams.reduce(
+		(acc: Record<number, Livestream[]>, video) => {
+			if (!acc[video.year]) acc[video.year] = []
+			acc[video.year].push(video)
+			return acc
+		},
+		{} as Record<number, Livestream[]>
+	)
+
+	// Sort years in descending order (most recent first)
+	const years = Object.keys(videosByYear)
+		.map(Number)
+		.sort((a, b) => b - a)
+
+	// Get the most recent video across all years
+	const allVideos = [...livestreams].sort(
```

Comment:
I think sort is expensive enough it warrants memoization for `allVideos` and `years`